packages/main/src/modules/config.ts

```diff
@@ -1,30 +1,38 @@
 import path from 'node:path';
-import { FileSystemStorage } from '/shared/types/storage';
+import { FileSystemStorage, type IFileSystemStorage } from '/shared/types/storage';
 import { SETTINGS_DIRECTORY, CONFIG_FILE_NAME, getFullScenesPath } from '/shared/paths';
-import { DEFAULT_CONFIG, mergeConfig } from '/shared/types/config';
+import { DEFAULT_CONFIG, mergeConfig, type Config } from '/shared/types/config';
 import { getUserDataPath } from './electron';
+import { waitForMigrations } from './migrations';
+import log from 'electron-log/main';
 
 export const CONFIG_PATH = path.join(getUserDataPath(), SETTINGS_DIRECTORY, CONFIG_FILE_NAME);
-const storage = await FileSystemStorage.getOrCreate(CONFIG_PATH);
 
-// Initialize with default values if empty
-const existingConfig = await storage.get<Record<string, any>>('');
-const defaultConfig = { ...DEFAULT_CONFIG };
-defaultConfig.settings.scenesPath = getFullScenesPath(getUserDataPath());
+let configStorage: IFileSystemStorage | undefined;
 
-if (!existingConfig || Object.keys(existingConfig).length === 0) {
-  // Write the default config
-  for (const [key, value] of Object.entries(defaultConfig)) {
-    await storage.set(key, value);
-  }
-} else {
-  // Deep merge with defaults if config exists but might be missing properties
-  const mergedConfig = mergeConfig(existingConfig, defaultConfig);
-  if (JSON.stringify(existingConfig) !== JSON.stringify(mergedConfig)) {
-    for (const [key, value] of Object.entries(mergedConfig)) {
-      await storage.set(key, value);
+export async function getConfig(): Promise<IFileSystemStorage> {
+  await waitForMigrations();
```

Comment:
This is just to avoid possible race conditions between the migration of the config and the initialization of the default values.